File: 30-days-of-code/cpp/day-09-recursion.cpp
#include <bits/stdc++.h>

using namespace std;

int factorial(int n) {
    // Begin Solution
    //---------------------------------------------------------------------------------------
    if (n == 1) {
        return 1;
    } else {
        return factorial(n-1) * n;
    };
    //---------------------------------------------------------------------------------------
    // End Solution
}

int main() {
    ofstream fout(getenv("OUTPUT_PATH"));

    int n;
    cin >> n;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');

    int result = factorial(n);

    fout << result << "\n";

    fout.close();

    return 0;
}

File: 30-days-of-code/php/day-09-recursion.php
<?php

function factorial($n) {
  // Begin Solution
  //---------------------------------------------------------------------------------------
  if($n == 1){
    return 1;
  } else {
    return factorial($n-1) * $n;
  };
  //---------------------------------------------------------------------------------------
  // End Solution
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$stdin = fopen("php://stdin", "r");

fscanf($stdin, "%d\n", $n);

$result = factorial($n);

fwrite($fptr, $result . "\n");

fclose($stdin);
fclose($fptr);
?>